File: CPPSolutions/CPPSolutions/Q887_鸡蛋掉落.cpp
#include <bits\stdc++.h>

using namespace std;

const int K = 110;

class Solution {
private:
    int f[K];

public:
    int superEggDrop(int k, int n) {
        if (n == 1) return 1;
        int cnt = 0;
        while (f[k] < n) {
            cnt++;
            for (int i = k; i >= 1; i--) 
                f[i] = f[i] + f[i - 1] + 1;
        }

        return cnt;
    }
};

File: GolangSolutions/DP/Knapsack/Q887_鸡蛋掉落.go
package Knapsack

func superEggDrop(k int, n int) (cnt int) {
	if n == 1 {
		return 1
	}
	f := make([]int, k+1)
	for f[k] < n {
		cnt++
		for i := k; i >= 1; i-- {
			f[i] = f[i] + f[i-1] + 1
		}
	}

	return
}
